src/main/java/org/opensearch/security/util/KeyUtils.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.util;
+
+import io.jsonwebtoken.JwtParser;
+import io.jsonwebtoken.Jwts;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.SpecialPermission;
+
+import java.security.AccessController;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivilegedAction;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.Objects;
+
+public class KeyUtils {
+
+    public static JwtParser createJwtParserFromSigningKey(final String signingKey, final Logger log) {
+        final SecurityManager sm = System.getSecurityManager();
+
+        JwtParser jwtParser = null;
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        jwtParser = AccessController.doPrivileged(new PrivilegedAction<JwtParser>() {
```

Comment:
Can we update this to match what is in this PR? https://github.com/opensearch-project/security/pull/3189

I noticed when wrapping the call to create the parser that many of the methods used in here are deprecated in JJWT. JJWT now recommends using a JWTParserBuilder: https://javadoc.io/static/io.jsonwebtoken/jjwt-api/0.11.2/index.html?io/jsonwebtoken/JwtParserBuilder.html

Reply:
Sounds good, I have updated the KeyUtils, OBOAuthenticator and JWTAuthenticator for using this JWTParserBuilder.